### PR TITLE
Fix: support name attribute for smtp plugin

### PR DESCRIPTION
**Pull request by DarkPhoenix2704**
## Change Summary

Provide summary of changes with issue number if any.

Closes https://github.com/nocodb/nocodb/issues/7103

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nocodb/src/plugins/smtp/SMTP.ts [hunk 1]
```diff
@@ -23,13 +24,11 @@ export default class SMTP implements IEmailAdapter {
...
           ? this.input?.ignoreTLS
           : this.input?.ignoreTLS === 'true',
-      auth: {
-        user: this.input?.username,
-        pass: this.input?.password,
-      },
       tls: {
         rejectUnauthorized: this.input?.rejectUnauthorized,
       },
+      ...(this.input?.user ? { auth: { user: this.input?.user } } : {}),
```
Comment (pranavxc) [inline packages/nocodb/src/plugins/smtp/SMTP.ts:30]: is it `user` or `username`  ?  As per plugin meta it's `username`

Comment (DarkPhoenix2704) [reply packages/nocodb/src/plugins/smtp/SMTP.ts:30]: image
It's user

Comment (pranavxc) [reply packages/nocodb/src/plugins/smtp/SMTP.ts:30]: `this.input?.user` => `this.input?.username`

File: packages/nocodb/src/plugins/smtp/SMTP.ts [hunk 2]
```diff
@@ -23,13 +24,11 @@ export default class SMTP implements IEmailAdapter {
...
           : this.input?.ignoreTLS === 'true',
-      auth: {
-        user: this.input?.username,
-        pass: this.input?.password,
-      },
       tls: {
         rejectUnauthorized: this.input?.rejectUnauthorized,
       },
+      ...(this.input?.username ? { auth: { user: this.input?.username } } : {}),
+      ...(this.input?.password ? { auth: { pass: this.input?.password } } : {}),
```
Comment (pranavxc) [inline packages/nocodb/src/plugins/smtp/SMTP.ts:31]: this is wrong if both are defined password will overwrite the other.

Should be something like:

```typescript
...(this.input?.username || this.input?.password ? { auth: { 
      ...(this.input?.username ? {user: this.input?.username } : {}),
      ...(this.input?.password ? { pass: this.input?.password } : {}),
})
```

Comment (DarkPhoenix2704) [reply packages/nocodb/src/plugins/smtp/SMTP.ts]: Missed them. Fixed in  [be61a25](https://github.com/nocodb/nocodb/pull/8412/commits/be61a25adfcd72fcc98b065d8056a8cceba75542)

**Review by pranavxc — APPROVED** [review]

**Pull request merged**